doc/DataPlaneCodeGeneration/Subclients.md

```diff
@@ -0,0 +1,335 @@
+# Customizing the generated code
+
+## Subclients
+
+Autorest requires each operation in swagger file to define a unique string parameter `operationId` (this is different from [official OpenAPI specification](https://swagger.io/docs/specification/paths-and-operations/#operationId)). If parameter value contains underscore, then the part after underscore will be treated as operation name, and the part before underscore will be treated as the name of the operation group which this operation belongs to. Otherwise, operation will be attributed to the group without a name::
+
+```js
+"paths": {
+  "/namedgroup/op": {
+    "get": {
+      "operationId": "NamedGroup_Op1",
+    }
+  },
+  "/nogroup/op": {
+    "get": {
+      "operationId": "Op2",
+    }
+  }
+}
+```
+
+For each operation group, **autorest.csharp** generates individual client type called using `<operationGroupName>Client` format when group has a name or `<serviceName>Client` format when group name is empty.
+
+```cs
+public partial class NamedGroupClient
+{
+    public virtual async Task<Response> Op1Async(RequestContext options = null);
+    public virtual Response Op1(RequestContext options = null);
+}
+```
+
+```cs
+public partial class MyServiceClient
+{
+    public virtual async Task<Response> Op2Async(RequestContext options = null);
+    public virtual Response Op2(RequestContext options = null);
+}
+```
+
+As above, one Azure service maybe be exposed to .NET developers as more than one clients that is not easy for .NET developers to discover and find out the right starting point to call Azure services with the Azure SDK. To resolve this, we can export only one service client in its main namespace as the main starting point for developers via [subclient feature](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients).
```

Comment:
Please put this before line 5.

Reply:
Those statements is based on the information of line 1-39. So I think it is not good to put before line 5.